eva_submission/eload_utils.py

```diff
@@ -183,4 +184,56 @@ def backup_file(file_name):
 def download_file(url, dest):
     """Download a public file accessible via http or ftp."""
     urllib.request.urlretrieve(url, dest)
-    urllib.request.urlcleanup()
\ No newline at end of file
+    urllib.request.urlcleanup()
+
+
+def get_vep_cache_version_from_ensembl(assembly_accession):
+    try:
+        logger.info(f'Getting vep_cache_version for assembly_accession : {assembly_accession}')
+        species_assembly = getSpeciesNameAndAssembly(assembly_accession)
+        logger.info(f'Details from Ensembl for species and assembly : {species_assembly}')
+
+        url = "ftp.ensembl.org"
+        ftp = ftplib.FTP(url)
+        ftp.login()
+
+        all_releases = get_releases(ftp)
+        logger.info(f'fetched all releases: {all_releases}')
+
+        for release in sorted(all_releases, reverse=True):
+            logger.info(f'looking for vep_cache_version in release : {all_releases.get(release)}')
+            all_species_files = ftp.nlst(all_releases.get(release) + "/variation/vep")
+            for file in all_species_files:
+                if species_assembly['species'] in file and species_assembly['assembly'] in file:
+                    logger.info(
+                        f'Found vep_cache_version for the species and assembly : {species_assembly}, file: {file}, release: {release}')
+                    return release
+        logger.info(f'could not find vep_cache_version for the given species and assembly: {species_assembly}')
+        return None
+    except Exception as err:
+        logger.info(f'Encountered Error while fetching vep_cache_version : {err}')
+        return None
```

Comment:
If we return `None` here, that means we'll skip annotation anytime there are issues with the FTP connection, right?  I think we wouldn't want to do that, e.g. if we already have the VEP cache downloaded so annotation should go ahead...

Reply:
(If we're only checking FTP when no cache version is passed in, as in my other comment, then this is less of a problem - maybe just make it a `warning` to highlight that we're skipping annotation because something weird happened.)

Reply:
Agree with @apriltuesday. Shall we annotate this method with "retries" so that occasional network disconnects don't ruin the results?

Reply:
moved the ftp and rest calls to separate individual methods and annotated all with retry logic.

Reply:
updated